launcher/container_runner.go

```diff
@@ -230,15 +259,29 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 		return nil, fmt.Errorf("failed to create REST verifier client: %v", err)
 	}
 
+	// Fetch container image signatures with specific repositories.
+	containerSignaturesFetcher := func(targetRepos []string) []internal.Signature {
+		sdClient := getSignatureDiscoveryClient(cdClient, token, image.Target())
```

Comment:
are we getting a new signaturediscovery Client every time we call agent.Attest? Can we make this a closure to avoid this? Alternatively, we can pass the signature discovery client in the agent and use a fake for testing.